diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97863ffb5..7163c8553 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,8 @@ set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum OS X deployment ve
 project(binaryen LANGUAGES C CXX VERSION 129)
 include(GNUInstallDirs)
 
+INCLUDE(eng/functions.cmake)
+
 # The C++ standard whose features are required to build Binaryen.
 # Keep in sync with scripts/test/shared.py cxx_standard
 # The if condition allows embedding in a project with a higher default C++ standard set
@@ -184,7 +186,7 @@ function(binaryen_add_executable name sources)
   target_link_libraries(${name} PRIVATE Threads::Threads)
   target_link_libraries(${name} PRIVATE binaryen)
   binaryen_setup_rpath(${name})
-  install(TARGETS ${name} DESTINATION ${CMAKE_INSTALL_BINDIR})
+  install_with_stripped_symbols(${name} TARGETS bin)
 endfunction()
 
 # Compiler setup. Use SYSTEM to avoid warnings and errors from third-party headers.
@@ -386,6 +388,7 @@ endif()
 # Configure warnings and errors
 
 if(MSVC)
+  add_compile_flag("/nologo")
   add_compile_flag("/wd4146") # Ignore warning "warning C4146: unary minus operator applied to unsigned type, result still unsigned", this pattern is used somewhat commonly in the code.
   # 4267 and 4244 are conversion/truncation warnings. We might want to fix these but they are currently pervasive.
   add_compile_flag("/wd4267")
@@ -407,6 +410,7 @@ else() # MSVC
   add_compile_flag("-Wall")
   if(ENABLE_WERROR)
     add_compile_flag("-Werror")
+    add_compile_flag("-Wno-error=unused-command-line-argument")
   endif()
   add_compile_flag("-Wextra")
   add_compile_flag("-Wno-unused-parameter")
@@ -518,10 +522,7 @@ set(binaryen_SOURCES
 target_sources(binaryen PRIVATE ${binaryen_SOURCES})
 
 if(INSTALL_LIBS OR NOT BUILD_STATIC_LIB)
-  install(TARGETS binaryen
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  install_with_stripped_symbols(binaryen TARGETS lib)
 endif()
 if(INSTALL_LIBS)
   install(FILES src/binaryen-c.h src/wasm-delegations.def DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
@@ -568,7 +569,7 @@ if(EMSCRIPTEN)
   # Don't exit the process on a fatal error, instead throw, so that JS can
   # catch.
   add_compile_flag("-DTHROW_ON_FATAL")
-  install(TARGETS binaryen_js DESTINATION ${CMAKE_INSTALL_BINDIR})
+  install_with_stripped_symbols(binaryen_js TARGETS bin)
 endif()
 
 configure_file(scripts/binaryen-lit.in ${CMAKE_BINARY_DIR}/bin/binaryen-lit @ONLY)
